packages/edit-site/src/components/sidebar-navigation-screen-page/index.js

```diff
@@ -18,42 +35,235 @@ import SidebarNavigationScreen from '../sidebar-navigation-screen';
+			value: <Truncate numberOfLines={ 1 }>{ page.slug }</Truncate>,
+		},
+	];
+
+	if ( page?.templateTitle ) {
+		details.push( {
+			label: __( 'Template' ),
+			value: page.templateTitle,
+		} );
+	}
+
+	details.push( {
+		label: __( 'Parent' ),
+		value: page?.parentTitle,
+	} );
+
+	/*
+	 * translators: If your word count is based on single characters (e.g. East Asian characters),
+	 * enter 'characters_excluding_spaces' or 'characters_including_spaces'. Otherwise, enter 'words'.
+	 * Do not translate into your own language.
+	 */
+	const wordCountType = _x( 'words', 'Word count type. Do not translate!' );
+	const wordsCounted = page?.content?.rendered
+		? wordCount( page.content.rendered, wordCountType )
+		: 0;
+	const readingTime = Math.round( wordsCounted / AVERAGE_READING_RATE );
+
+	if ( wordsCounted ) {
+		details.push(
+			{
+				label: __( 'Words' ),
+				value: wordsCounted.toLocaleString() || __( 'Unknown' ),
+			},
+			{
+				label: __( 'Time to read' ),
+				value:
+					readingTime > 1
+						? sprintf(
+								/* translators: %s: is the number of minutes. */
+								__( '%s mins' ),
+								readingTime.toLocaleString()
+						  )
+						: __( '< 1 min' ),
+			}
+		);
+	}
+	return details;
+}
 
 export default function SidebarNavigationScreenPage() {
 	const { setCanvasMode } = unlock( useDispatch( editSiteStore ) );
+	const { getFeaturedMediaDetails } = unlock( privateEditorApis );
 	const {
 		params: { postId },
 	} = useNavigator();
 	const { record } = useEntityRecord( 'postType', 'page', postId );
 
-	return (
+	const {
+		parentTitle,
+		featuredMediaDetails: { mediaSourceUrl, altText },
+		templateSlug,
+	} = useSelect(
+		( select ) => {
+			const { getEditedPostContext } = unlock( select( editSiteStore ) );
+
+			// Featured image.
+			const attachedMedia = record?.featured_media
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'attachment',
+						record?.featured_media
+				  )
+				: null;
+
+			// Parent page title.
+			let _parentTitle = record?.parent
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'page',
+						record.parent,
+						{ _fields: [ 'title' ] }
+				  )
+				: null;
+
+			if ( _parentTitle?.title ) {
+				_parentTitle = _parentTitle.title?.rendered
+					? decodeEntities( _parentTitle.title.rendered )
+					: __( 'Untitled' );
+			} else {
+				_parentTitle = __( 'Top level' );
+			}
+
+			return {
+				parentTitle: _parentTitle,
+				templateSlug: getEditedPostContext()?.templateSlug,
+				featuredMediaDetails: {
+					...getFeaturedMediaDetails( attachedMedia, postId ),
+					altText: escapeAttribute(
+						attachedMedia?.alt_text ||
+							attachedMedia?.description?.raw ||
+							''
+					),
+				},
+			};
+		},
+		[ record, postId ]
+	);
+
+	// Match template slug to template title.
+	const { records: templates, isResolving: areTemplatesLoading } =
+		useEntityRecords( 'postType', 'wp_template', {
+			per_page: -1,
+		} );
```

Comment:
Do we really need to fetch all the templates, to retrieve one by slug, can't we call a single `getEntityRecord` instead?

Reply:
I had checked this too. We cannot make a request for templates by `slug` if I'm not wrong. That's something we should address though at some point.

Reply:
I think we can, for templates the `theme//slug` is the id

Reply:
I also tried `getEntityRecord` and it does return `slug`, but only the template name segment, e.g., 

```js
{
    "postType": "page",
    "postId": "2",
    "templateSlug": "page"
}
```

So I needed a way to get the current theme to build the id, e.g., `twentytwentythree//${ _templateSlug }`

Reply:
And now I've just discoved `getCurrentTheme()` :trollface:

Reply:
Nice! 

That's the first place I looked but it's using `availableTemplates`, which is a property of `select( editorStore ).getEditorSettings()`, but now I see the site editor settings have it under `defaultTemplateTypes`. 👍 Updated.